### PR TITLE
Makefile: Enable additional warnings, but don't yet make warnings fatal.

### PR DESCRIPTION
**Pull request by AlexanderKurtz**
Since we want this not only for liblaik itself, but also for all other parts
of LAIK, this commit also changes the build system to properly pass CFLAGS,
CXXFLAGS and LDFLAGS to recursive make calls, which has the nice side effect
of making the other Makefiles more readable and also streamlines the compiler
options across the project (as far as possible).

### CODE REVIEW

**Comment by AlexanderKurtz** [comment]
Huh, Travis didn't fail? Maybe it's GCC is so ancient that it doesn't check for ```-Wunused-parameters```?

**Comment by AlexanderKurtz** [comment]
Travis currently fails because clang doesn't (yet) know ```-Wno-unused-but-set-variable```. However, the gcc tests all pass and once #25 and #26 are merged, this exception will no longer be necessary.

File: examples/raytracer.cpp [hunk 1]
```diff
@@ -43,7 +43,9 @@
 #include <cstring>
 
 extern "C"{
+#ifndef USE_MPI
```
Comment (weidendo) [inline examples/raytracer.cpp:46]: I do not understand this. If MPI is not available, force usage of MPI obviously cannot work. So what's the point?

Comment (AlexanderKurtz) [reply examples/raytracer.cpp:46]: No idea. @twittidai? However, redefining a macro triggers a warning here, so I added an ```ifdef```

**Comment by weidendo** [comment]
I understand the idea, but this PR has a flaw: now if you run 'make' in a subdirectory, CFLAGS etc. will not be correctly set. Actually, this PR would change the Makefile's back to what they were some time ago...

Instead of "streamlining", I much more prefer Makefile's to be obvious about what they do, even if this involves some redundant code. I actually think that code which tries hard to eliminate all redundancy is misguided: it is much more difficult to understand and thus harmful.

Enabling warnings of course is a good thing.
Every Makefile could start with including a top-level "Makefile.def" to set global defaults, which also would set the WARNINGS. Then Makefile.config eventually can overwrite this.

**Comment by AlexanderKurtz** [comment]
I just pushed a new version as #42. make-ing in a subdirectory should work now, but please note that dependency tracking is just as broken as before, i.e. make-ing the examples without first make-ing liblaik won't work.

**Pull request closed without merging**